### ISSUE TITLE
TimerQueue.DeleteAppDomainTimer sometimes failing during finalization

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#1841]
This has been happening periodically while running our tests in CI.
```
  Unhandled Exception: System.Exception: The handle is invalid. (Exception from HRESULT: 0x80070006 (E_HANDLE))
     at System.Threading.TimerQueue.DeleteAppDomainTimer(IntPtr handle)
     at System.Threading.TimerQueue.AppDomainTimerSafeHandle.ReleaseHandle()
     at System.Runtime.InteropServices.SafeHandle.InternalFinalize()
     at System.Runtime.InteropServices.SafeHandle.Dispose(Boolean disposing)
     at System.Runtime.InteropServices.SafeHandle.Finalize()
```

cc: @ericeil... Eric, have you seen this before?

**Comment by bpschoch** [comment]
I've gotten these while testing pipes.  It maybe related to pipes but don't know how to really debug it when I occasionally get this error.  Some of the pipes testing does use read timeouts and cancellation tokens.

**Comment by stephentoub** [comment]
It's possible this is actually being caused by a handle recycling problem, fixed in #1871.  I'll close this once #1871 is merged in, and we can reopen it subsequently if we see it again.

**Comment by stephentoub** [comment]
I believe #1871 fixes this.  Closing for now; we can reopen if we see this again.
